src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerDelegate.scala

```diff
@@ -114,4 +109,15 @@ private[tracker] class InstanceTrackerDelegate(
     import scala.concurrent.ExecutionContext.Implicits.global
     process(InstanceUpdateOperation.ReservationTimeout(instanceId)).map(_ => Done)
   }
+
+  @SuppressWarnings(Array("all")) /* async/await */
+  override def instancesToScheduleCount(appId: PathId, targetCount: Int): Future[Int] = {
```

Comment:
I'm not so sure about the benefit. However, we could make this more generic and count based on a provided filter

```scala
def count(specId: PathId, filter: (Instance) => Boolean): Future[Int] = async {
  val instances = await(list(specId))
  instances.count(filter)
}
```

Reply:
my main motivation for this was:
- the list of possible states to filter when trying to figure out how much to scales is growing, right now this computation was on two places and those two lists were not in sync (possibly causing some regressions)
- in the future this method should go anyway - we count how much to schedule only to call instancetracker.schedule in the end, so I think that in the future there should be just `schedule(appId, targetCount)` method that will schedule (or not schedule) based on the instance tracker state. But to do this, we have to get rid of LaunchQueue first

So this is really just putting the algorithm to compute how much to schedule to one place inside instance tracker...

Reply:
Good points! I'm convinced.